### ISSUE TITLE
tell ./configure script to look in a different directory?

### ISSUE THREAD

**Issue by toddfraser88** [joncampbell123/dosbox-x#223]
is there a way to tell ./configure script to look in a different directory?

I mean, have it look in a different location for things like pcap, fluidsynth, SDL?

I'm just asking so I might be able to compile 32-bit dosbox-x on 64-bit linux, without having to replace 64-bit SDL with 32-bit SDL on the machine

**Comment by joncampbell123** [comment]
The configure script uses AC_CHECK_HEADER and AC_CHECK_LIB, so I would assume that if you wanted it to look elsewhere first you could first set environment variables LDFLAGS, CXXFLAGS, CPPFLAGS, and CFLAGS with directives to first search the include and lib directories you want:

export LDFLAGS=-L/usr/lib32
export CFLAGS=-I/usr/include32
export CPPFLAGS=-I/usr/include32
export CXXFLAGS=-I/usr/include32

Some code in configure.ac also uses pkg-config, so you'll need to set PKG_CONFIG_PATH as well to point to the right dir:

export PKG_CONFIG_PATH=/usr/lib32/pkgconfig

**Comment by toddfraser88** [comment]
thanks for that, ./configure now works, 

but make seems to fail at the dynamic core stage even with the environment variables set

core_dyn_x86/dyn_fpu.h:616:79: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘Bitu {aka long unsigned int}’ [-Wformat=]
core_dyn_x86/dyn_fpu.h:621:77: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘Bitu {aka long unsigned int}’ [-Wformat=]
    LOG(LOG_FPU,LOG_WARN)("ESC 7:Unhandled group %d subfunction %d",group,sub);
                                                                             ^
core_dyn_x86/dyn_fpu.h:621:77: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘Bitu {aka long unsigned int}’ [-Wformat=]
core_dyn_x86/dyn_fpu.h:633:80: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘Bitu {aka long unsigned int}’ [-Wformat=]
    LOG(LOG_FPU,LOG_WARN)("ESC 7 EA:Unhandled group %d subfunction %d",group,sub);
                                                                                ^
core_dyn_x86/dyn_fpu.h:633:80: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘Bitu {aka long unsigned int}’ [-Wformat=]
core_dyn_x86/dyn_fpu.h:661:80: warning: format ‘%d’ expects argument of type ‘int’, but argument 3 has type ‘Bitu {aka long unsigned int}’ [-Wformat=]
    LOG(LOG_FPU,LOG_WARN)("ESC 7 EA:Unhandled group %d subfunction %d",group,sub);
                                                                                ^
core_dyn_x86/dyn_fpu.h:661:80: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘Bitu {aka long unsigned int}’ [-Wformat=]
In file included from core_dyn_x86.cpp:234:0:
core_dyn_x86/decoder.h: In function ‘CacheBlock* CreateCacheBlock(CodePageHandler*, PhysPt, Bitu)’:
core_dyn_x86/decoder.h:1979:59: error: cast from ‘CacheBlock*’ to ‘Bit32u {aka unsigned int}’ loses precision [-fpermissive]
  gen_save_host_direct(&cache.block.running,(Bit32u)decode.block);
                                                           ^
Makefile:396: recipe for target 'core_dyn_x86.o' failed

**Comment by joncampbell123** [comment]
If I understand Linux GCC correctly, you might try adding -m32 to the CFLAGS to tell GCC you're trying to compile 32-bit.

**Comment by toddfraser88** [comment]
i tried

sudo make clean
export CFLAGS=-m32
sudo make

unfortunately, same error

**Comment by joncampbell123** [comment]
Yeah, that method of cross compiling never worked for me either. :)

My recommendation then is to do one of two things.

1) Install VirtualBox or QEMU, install a 32-bit Linux inside it, and then compile it there.

2) Install a 32-bit Linux on another machine, then copy the system root to a subdirectory on your 64-bit Linux. Then when you need to compile DOSBox-X as 32-bit, sudo as root on your 64-bit Linux, then chroot into that directory. At that point you have a 32-bit Linux subsystem inside 64-bit (don't expect much besides basic tools to run though) and you should be able to compile.

**Comment by toddfraser88** [comment]
> 1) Install VirtualBox or QEMU, install a 32-bit Linux inside it, and then compile it there.

Whilst it's possible to get dosbox-x to compile as 32-bit on 64-bit, you end up having to install 32-bit versions of it's dependencies, which uninstalls the 64-bit versions.

You're right, It seems booting into a 32-bit linux and compiling it there is the best option.
